cpp/benchmarks/abm.cpp

```diff
@@ -0,0 +1,157 @@
+#include "abm/simulation.h"
+#include "memilio/utils/stl_util.h"
+#include "benchmark/benchmark.h"
+
+mio::abm::Simulation make_simulation(size_t num_persons, std::initializer_list<uint32_t> seeds)
+{
+    auto rng = mio::RandomNumberGenerator();
+    rng.seed(seeds);
+    auto world      = mio::abm::World(5);
+    world.get_rng() = rng;
+
+    //create persons at home
+    const auto mean_home_size    = 5.0;
+    const auto min_home_size     = 1;
+    auto& home_size_distribution = mio::PoissonDistribution<int>::get_instance();
+    auto home                    = world.add_location(mio::abm::LocationType::Home);
+    auto planned_home_size       = home_size_distribution(world.get_rng(), mean_home_size);
+    auto home_size               = 0;
+    for (size_t i = 0; i < num_persons; ++i) {
+        if (home_size >= std::max(min_home_size, planned_home_size)) {
+            home              = world.add_location(mio::abm::LocationType::Home);
+            planned_home_size = home_size_distribution(world.get_rng(), mean_home_size);
+            home_size         = 0;
+        }
+
+        auto age     = mio::AgeGroup(mio::UniformIntDistribution<size_t>::get_instance()(
+            world.get_rng(), size_t(0), world.parameters.get_num_groups() - 1));
+        auto& person = world.add_person(home, age);
+        person.set_assigned_location(home);
+        home_size++;
+    }
+
+    //create other locations
+    for (auto loc_type :
+         {mio::abm::LocationType::School, mio::abm::LocationType::Work, mio::abm::LocationType::SocialEvent,
+          mio::abm::LocationType::BasicsShop, mio::abm::LocationType::Hospital, mio::abm::LocationType::ICU}) {
+
+        const auto num_locs = std::max(size_t(1), num_persons / 2'000);
+        std::vector<mio::abm::LocationId> locs(num_locs);
+        std::generate(locs.begin(), locs.end(), [&] {
+            return world.add_location(loc_type);
+        });
+        for (auto& person : world.get_persons()) {
+            auto loc_idx =
+                mio::UniformIntDistribution<size_t>::get_instance()(world.get_rng(), size_t(0), num_locs - 1);
+            person.set_assigned_location(locs[loc_idx]);
+        }
+    }
+
+    //infections and masks
+    for (auto& person : world.get_persons()) {
+        auto prng = mio::abm::Person::RandomNumberGenerator(world.get_rng(), person);
+        //~1% of people are infected, large enough to have some infection activity without everyone dying
+        if (mio::UniformDistribution<double>::get_instance()(prng, 0.0, 1.0) < 0.05) {
```

Comment:
You say about 1% of people are infected, but then number is 0.05. Which one did you want to use?

Reply:
that's why you don't repeat a value in the comments...

I removed the number from the comment and introduced a variable that contains the percentage instead. Also did to some other places in the file.